edx_proctoring/api.py

```diff
@@ -1222,55 +1325,74 @@ def _get_proctored_exam_view(exam, context, exam_id, user_id, course_id):
-        credit_state = context['credit_state']
-
+    if check_mode:
         has_mode = _check_eligibility_of_enrollment_mode(credit_state)
-        has_prerequisites = False
-        if has_mode:
-            has_prerequisites = _check_eligibility_of_prerequisites(credit_state)
-
-        # see if the user has passed all pre-requisite credit eligibility
-        # checks, otherwise just show the user the exam unproctored
-        if not has_mode or not has_prerequisites:
-            # if we are in the right mode and if we don't have
-            # pre-requisites, then we implicitly decline the exam
-            if has_mode:
-                attempt = get_exam_attempt(exam_id, user_id)
-                if not attempt:
-                    # user hasn't a record of attempt, create one now
-                    # so we can mark it as declined
-                    create_exam_attempt(exam_id, user_id)
-
-                update_attempt_status(
-                    exam_id,
-                    user_id,
-                    ProctoredExamStudentAttemptStatus.declined,
-                    raise_if_not_found=False
-                )
-
-            # don't override context, let the courseware show
+        if not has_mode:
+            # user does not have the required enrollment mode
+            # so do not override view this is a quick exit
             return None
+
     attempt = get_exam_attempt(exam_id, user_id)
 
     attempt_status = attempt['status'] if attempt else None
 
     # if user has declined the attempt, then we don't show the
-    # proctored exam
+    # proctored exam, a quick exit....
     if attempt_status == ProctoredExamStudentAttemptStatus.declined:
         return None
 
     if not attempt_status:
-        student_view_template = 'proctored_exam/entrance.html'
+        # student has not started an attempt
+        # so, show them:
+        #       1) If there are failed prerequisites then block user and say why
+        #       2) If there are pending prerequisites then block user and allow them to remediate them
+        #       3) Otherwise - all prerequisites are satisfied - then give user
+        #          option to take exam as proctored
+
+        # get information about prerequisites
+
+        credit_requirement_status = (
+            credit_state.get('credit_requirement_status')
+            if credit_state else []
+        )
+
+        prerequisite_status = _are_prerequirements_satisfied(
+            credit_requirement_status,
+            evaluate_for_requirement_name=exam['content_id'],
+            filter_out_namespaces=['grade']
+        )
+
+        # add any prerequisite information, if applicable
+        context.update({
+            'prerequisite_status': prerequisite_status
+        })
+
+        if not prerequisite_status['are_prerequisites_satisifed']:
+            # do we have failed prerequisites? That takes priority
+            if prerequisite_status['failed_prerequisites']:
```

Comment:
Yes, because if you have failed prereqs, there's little an end user can do outside of contacting Support if they want to contest a failure.